indexer/feature_utils.cpp

```diff
@@ -221,41 +202,78 @@ int GetFeatureViewportScale(TypesHolder const & types)
   return impl::GetFeatureEstimator().GetViewportScale(types);
 }
 
-void GetPreferredNames(FeatureID const & id, StringUtf8Multilang const & src, string & primary,
-                       string & secondary)
+// Primary name using priority:
+// - device language name;
+// - international name;
+// - english name.
+// Secondary name using priority:
+// - default name;
+// - international name;
+// - country language name;
+// - english name.
+void GetPreferredNames(FeatureID const & id, StringUtf8Multilang const & src,
+                       int8_t const deviceLang, string & primary, string & secondary)
 {
-  // Primary name using priority:
-  // - default name;
-  // - country language name.
-  // Secondary name using priority:
-  // - device language name;
-  // - international name;
-  // - english name.
-  GetNames(id, src, primary, secondary);
+  using StrUtf8 = StringUtf8Multilang;
+  primary.clear();
+  secondary.clear();
+
+  if (src.IsEmpty())
+    return;
+
+  vector<int8_t> const primaryCodes = {deviceLang,
+                                       StrUtf8::kInternationalCode,
+                                       StrUtf8::kEnglishCode};
+  vector<int8_t> const secondaryCodes = {StrUtf8::kDefaultCode,
+                                        StrUtf8::kInternationalCode};
+
+  GetBestName(src, primaryCodes, primary);
+  GetBestName(src, secondaryCodes, secondary);
+
+  if (secondary.empty())
+    GetMwmLangName(id, src, secondary);
+  if (secondary.empty())
+    src.GetString(StrUtf8::kEnglishCode, secondary);
 
   if (primary.empty())
-  {
-    primary.swap(secondary);
-  }
-  else
-  {
-    // Filter out similar intName.
-    if (!secondary.empty() && primary.find(secondary) != string::npos)
-      secondary.clear();
-  }
+    primary.swap(secondary);  
```

Comment:
So, it's possible that primary name will be set to mwm language name. But I don't see country language name in the primary name priority list.